libs/pika/async_mpi/include/pika/async_mpi/mpi_helpers.hpp

```diff
@@ -54,24 +55,22 @@ namespace pika::mpi::experimental::detail {
         std::invoke_result_t<F, std::add_lvalue_reference_t<std::decay_t<Ts>>..., MPI_Request*>>;
 
     // -----------------------------------------------------------------
-    // return a scheduler on the mpi pool, with or without stack
-    inline auto mpi_pool_scheduler(execution::thread_priority p, bool stack = true)
+    // return a scheduler on the default pool with added priority if requested
+    inline auto default_pool_scheduler(execution::thread_priority p)
     {
-        ex::thread_pool_scheduler sched{&resource::get_thread_pool(get_pool_name())};
-        if (!stack)
-        {
-            sched = ex::with_stacksize(std::move(sched), execution::thread_stacksize::nostack);
-        }
-        sched = ex::with_priority(std::move(sched), p);
-        return sched;
+        return ex::with_priority(
+            ex::thread_pool_scheduler{
+                &pika::detail::get_runtime_ptr()->get_thread_manager().default_pool()},
+            p);
     }
 
     // -----------------------------------------------------------------
-    // return a scheduler on the default pool with added priority if requested
-    inline auto default_pool_scheduler(execution::thread_priority p)
+    // return a scheduler on the mpi pool, with or without stack
+    inline auto mpi_pool_scheduler(execution::thread_priority p)
     {
+        if (!pool_exists()) return default_pool_scheduler(p);
         return ex::with_priority(
-            ex::thread_pool_scheduler{&resource::get_thread_pool("default")}, p);
+            ex::thread_pool_scheduler{&resource::get_thread_pool(get_pool_name())}, p);
```

Comment:
Just checking: why does this not care about stack/no stack anymore? Is that intentional?

Reply:
I removed the option to use `no_stack` because it wasn't implemented properly everywhere and it made things more complicated. I can put it back in, but where should we decide if a task is created `no_stack` or not. if we transfer a task to the mpi pool for dispatch, then probably yes `no_stack` makes sense - but when we trigger a continuation it might not. I will add back no_stack in another PR after thinking more. (I'll clean up the comments in this PR)

Reply:
Thanks, that makes sense. Let's leave it out from here at least.